receiver/mongodbatlasreceiver/receiver.go

```diff
@@ -77,47 +81,106 @@ func (s *mongodbatlasreceiver) shutdown(context.Context) error {
+		}
+	}
 
-				if err := s.extractProcessDatabaseMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
-					return fmt.Errorf("error when polling process database metrics from MongoDB Atlas for process %s: %w", process.ID, err)
-				}
+	return nil
+}
 
-				if err := s.extractProcessDiskMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
-					return fmt.Errorf("error when polling process disk metrics from MongoDB Atlas for process %s: %w", process.ID, err)
-				}
-			}
+func (s *mongodbatlasreceiver) processProject(ctx context.Context, time timeconstraints, project *mongodbatlas.Project, projectCfg *ProjectConfig) error {
+	org, err := s.client.GetOrganization(ctx, project.OrgID)
+	if err != nil {
+		return fmt.Errorf("error retrieving organization: %w", err)
+	}
+
+	nodeClusterMap, providerMap, err := s.getNodeClusterNameMap(ctx, project.ID)
+	if err != nil {
+		return fmt.Errorf("error collecting clusters from project %s: %w", project.ID, err)
+	}
+
+	processes, err := s.client.Processes(ctx, project.ID)
+	if err != nil {
+		return fmt.Errorf("error retrieving MongoDB Atlas processes for project %s: %w", project.ID, err)
+	}
+
+	for _, process := range processes {
+		clusterName := nodeClusterMap[process.UserAlias]
+		providerValues := providerMap[clusterName]
+
+		if !shouldProcessCluster(projectCfg, clusterName) {
+			// Skip processing for this cluster
+			continue
+		}
+
+		if err := s.extractProcessMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
+			return fmt.Errorf("error when polling process metrics from MongoDB Atlas for process %s: %w", process.ID, err)
+		}
+
+		if err := s.extractProcessDatabaseMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
+			return fmt.Errorf("error when polling process database metrics from MongoDB Atlas for process %s: %w", process.ID, err)
+		}
+
+		if err := s.extractProcessDiskMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
+			return fmt.Errorf("error when polling process disk metrics from MongoDB Atlas for process %s: %w", process.ID, err)
 		}
 	}
+
 	return nil
 }
 
+// shouldProcessCluster checks whether a given cluster should be processed based on the project configuration.
+func shouldProcessCluster(projectCfg *ProjectConfig, clusterName string) bool {
+	// If there are no included clusters specified, process all clusters unless the is an excluded clusters.
+	if len(projectCfg.IncludeClusters) == 0 {
+		// Check if cluster is excluded.
+		_, isExcluded := projectCfg.excludesByClusterName[clusterName]
+		return !isExcluded
+	}
+
+	// If included clusters are specified, only process clusters that are included and not excluded.
+	_, isIncluded := projectCfg.includesByClusterName[clusterName]
+	_, isExcluded := projectCfg.excludesByClusterName[clusterName]
+	return isIncluded && !isExcluded
+}
+
```

Comment:
Is this necessarily related to including/excluding projects? Can it be a separate PR?

Reply:
Yes? This PR is adding a project config which is reusing an existing project config pattern that includes a project name, exclude and include clusters slice. I think it is reasonable for all the project config fields to be implemented if it is possible.